src/api/firebase.js

```diff
@@ -131,3 +131,36 @@ export async function checkItem(listId) {
 	const existingList = await getDocs(listCollectionRef);
 	return existingList.empty ? false : true;
 }
+
+export async function comparePurchaseUrgency(data) {
```

Comment:
some really cool logic in here! 🤘

But I think we can DRY this up a little bit. I noticed that you're filtering through the data twice, once to find `inactive`, and once to find `active`. We could update to map through the items one time, and push each item into a new array depending on which condition is applicable:

```
export async function comparePurchaseUrgency(data) {
	const today = new Date();
	const inactiveList = [];
	const activeList = [];
	data.map((item) => {
		if (item.dateLastPurchased !== null && numOfDaysBtwnDates(item.dateLastPurchased.toDate(), today) >= 60){
			inactiveList.push(item);
		} else if (item.dateLastPurchased === null || numOfDaysBtwnDates(item.dateLastPurchased.toDate(), today) < 60){
			activeList.push(item);
		}
	})
	return sortByUrgencyAndName(activeList).concat(sortByUrgencyAndName(inactiveList))
}
```

Reply:
This is way cleaner, thank you!  😎

Reply:
anytime :)